resolver/resolver.go

```diff
@@ -72,6 +72,10 @@ func (r *Resolver) ResolveToLastNode(ctx context.Context, fpath path.Path) (cid.
 	// create a selector to traverse and match all path segments
 	pathSelector := pathAllSelector(p[:len(p)-1])
 
+	// create a new cancellable session
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+
```

Comment:
This also seems to be the one that's used by go-ipfs and caused the reported error.

Reply:
Perhaps for all the other ones aside from this we just remove the timeout.

@Stebalien you were thinking that leaving these timeouts in might make debugging changes here easier (https://github.com/ipfs/go-path/pull/36#discussion_r675739954). Any objection to leaving this one in and removing the rest?

The change here means we have 1min to resolve the function instead of 1min per block, but that should be way more than enough.

Reply:
I can't remember why that had anything to do with tracking down bugs... (might have been WRT keeping changes minimal?).

I'm generally against timeouts in places like this where there's no reasonable "we should have received a response by X time" so I'd rather remove all of the hard-coded timeouts like this eventually.